sds_data_manager/lambda_code/SDSCode/api_lambdas/download_api.py

```diff
@@ -112,6 +159,16 @@ def lambda_handler(event, context):
             ),
         )
 
+    is_auth_user = is_authenticated_path(event)
+    file_is_released = is_released(filepath)
+
+    if not is_auth_user and not file_is_released:
```

Comment:
We don't want to query the DB unnecessarily, so if we are on the authenticated path, we don't care to look up whether it is released or not. If you put these into the if-block itself it won't evaluate the second condition and short-circuit to avoid the unnecessary query.
```suggestion
    if not is_authenticated_path(event) and not is_released(filepath):
```